azure-devops-client/main.go

```diff
@@ -181,14 +179,13 @@ func (c *AzureDevopsClient) restVsrm() *resty.Client {
 	return client
 }
 
-func (c *AzureDevopsClient) restWithAuthentication(restClient **resty.Client, domain string) (*resty.Client, error) {
-
-	if *restClient == nil {
-		*restClient = c.restWithoutToken(domain)
+func (c *AzureDevopsClient) restWithAuthentication(restClient *resty.Client, domain string) (*resty.Client, error) {
+	if restClient == nil {
```

Comment:
@mblaschke I see that you removed the double pointer I added here. I added this on purpose: assigning a value to `restClient` here will not update `c.restClient` that gets passed into this function. To make this work, I added the double pointer, so that I can actually update the `c.restClient` field.

Therefore, the client will not get cached in the devops client object (`c.restClient` will always be `nil`), `if restClient == nil` statement will always evaluate to `true`, and the client will always get reinitialized.